flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/scalar/ToTimestampLtzFunction.java

```diff
@@ -89,46 +94,41 @@ public ToTimestampLtzFunction(SpecializedFunction.SpecializedContext context) {
     }
 
     public @Nullable TimestampData eval(Number epoch) {
-        if (epoch == null) {
-            return null;
-        }
-        if (epoch instanceof Float || epoch instanceof Double) {
-            return DateTimeUtils.toTimestampData(epoch.doubleValue());
-        }
-        return DateTimeUtils.toTimestampData(epoch.longValue());
+        return eval(epoch, DEFAULT_PRECISION);
     }
 
-    public TimestampData eval(DecimalData epoch) {
-        if (epoch == null) {
-            return null;
-        }
-
-        return DateTimeUtils.toTimestampData(epoch);
+    public @Nullable TimestampData eval(DecimalData epoch) {
+        return eval(epoch, DEFAULT_PRECISION);
     }
 
     public @Nullable TimestampData eval(StringData timestamp) {
         if (timestamp == null) {
             return null;
         }
 
-        return DateTimeUtils.parseTimestampData(timestamp.toString());
+        return parseTimestampData(timestamp.toString());
     }
 
     public @Nullable TimestampData eval(StringData timestamp, StringData format) {
         if (timestamp == null || format == null) {
             return null;
         }
 
-        return DateTimeUtils.parseTimestampData(timestamp.toString(), format.toString());
+        return parseTimestampData(timestamp.toString(), format.toString());
     }
 
     public @Nullable TimestampData eval(
-            StringData timestamp, StringData format, StringData timezone) {
-        if (timestamp == null || format == null || timezone == null) {
```

Comment:
move the logic into from `DateTimeUtils.java` to `ToTimestampLtzFunction.java` to reduce redundant function calling.